### PR TITLE
Add link to Max Shards Per Node exception message

### PR DESCRIPTION
**Pull request by stefnestor**
👋🏽 howdy, team! 

Trying to follow ~~https://github.com/elastic/elasticsearch/pull/105607~~ https://github.com/elastic/elasticsearch/pull/110911, this links common error like ... to the [related doc](https://www.elastic.co/guide/en/elasticsearch/reference/7.14/size-your-shards.html#_this_action_would_add_x_total_shards_but_this_cluster_currently_has_yz_maximum_shards_open) which is common support volume
```
Elasticsearch exception [type=validation_exception, reason=Validation Failed: 1: this action would add [2] total shards, but this cluster currently has [1000]/[1000] maximum shards open;]

Validation Failed: 1: this action would add [x] shards, but this cluster currently has [y]/[y] maximum shards open | maximum normal shards open
```

### CODE REVIEW

File: docs/changelog/110993.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pr: 110993
+summary: (Err>Doc) Max Shards Per Node
```
Comment (DaveCTurner) [inline docs/changelog/110993.yaml:2]: This summary will appear in the release notes - could you edit it (and ideally the PR title) to make it more release-notes-worthy?

File: server/src/main/resources/org/elasticsearch/common/reference-docs-links.json [hunk 2]
```diff
@@ -38,5 +38,6 @@
   "NETWORK_BINDING_AND_PUBLISHING": "modules-network.html#modules-network-binding-publishing",
   "SNAPSHOT_REPOSITORY_ANALYSIS": "repo-analysis-api.html",
   "S3_COMPATIBLE_REPOSITORIES": "repository-s3.html#repository-s3-compatible-services",
-  "LUCENE_MAX_DOCS_LIMIT": "size-your-shards.html#troubleshooting-max-docs-limit"
+  "LUCENE_MAX_DOCS_LIMIT": "size-your-shards.html#troubleshooting-max-docs-limit",
+  "MAX_SHARDS_PER_NODE": "size-your-shards.html#_this_action_would_add_x_total_shards_but_this_cluster_currently_has_yz_maximum_shards_open"
```
[the remark this answers is not in the document]
Comment (stefnestor) [reply server/src/main/resources/org/elasticsearch/common/reference-docs-links.json:42]: Sure! Can you tell me what that means or share an example?

[the remark this answers is not in the document]
Comment (DaveCTurner) [reply server/src/main/resources/org/elasticsearch/common/reference-docs-links.json:42]: like the `[[troubleshooting-max-docs-limit]]` line added here: https://github.com/elastic/elasticsearch/pull/110911/files#diff-0f4a16463295656000a67419cdd19461596d33690c2cc010f6ce65b19d465c01R549

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Pull request merged**